handlers/api.go

```diff
@@ -1620,6 +1622,135 @@ func ApiValidatorIncomeDetailsHistory(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// ApiValidatorWithdrawals godoc
+// @Summary Get the withdrawal history of up to 100 validators for the last 100 epochs. To receive older withdrawals modify the epoch paraum
+// @Tags Validator
+// @Produce  json
+// @Param  indexOrPubkey path string true "Up to 100 validator indicesOrPubkeys, comma separated"
+// @Param  epoch query int false "the start epoch for the withdrawal history (default: latest epoch)"
+// @Success 200 {object} types.ApiResponse{data=[]types.ApiValidatorWithdrawalResponse}
+// @Failure 400 {object} types.ApiResponse
+// @Router /api/v1/validator/{indexOrPubkey}/withdrawals [get]
+func ApiValidatorWithdrawals(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	vars := mux.Vars(r)
+	maxValidators := getUserPremium(r).MaxValidators
+
+	queryIndices, err := parseApiValidatorParamToIndices(vars["indexOrPubkey"], maxValidators)
+	if err != nil {
+		sendErrorResponse(w, r.URL.String(), err.Error())
+		return
+	}
+
+	if len(queryIndices) == 0 {
+		sendErrorResponse(w, r.URL.String(), "no or invalid validator indicies provided")
+	}
+
+	q := r.URL.Query()
+
+	latestEpoch := services.LatestEpoch()
+	epoch, err := strconv.ParseUint(q.Get("epoch"), 10, 64)
+	if err != nil {
+		epoch = latestEpoch
+	}
+
+	endEpoch := epoch - 100
```

Comment:
Out of interest, why are only 100 epochs checked? I guess, this is because of performance reasons.

One minor detail though: We don't only check the last 100 epochs as stated via the swagger comments above but the last 101. The start and end epochs are both inclusive [HERE](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/1833/files#diff-30c25f2a7d8ae335d2aada4b0a6013ad9d159595bb6a2a14b1d0022451b6753fR2615).

Thoughts?

Reply:
There is a query parameter epoch with which you can view historical epochs as well. To limit the response size at most 100 epochs are returned. We have 100 validators that are possible in a single query and 100 epochs, which would return in the worst case an array with around 100 results

Reply:
Yes its inclusive and returns 101 results, can be fixed